sympy/solvers/decompogen.py

```diff
@@ -37,13 +37,28 @@ def decompogen(f, symbol):
     if isinstance(f, (Function, Pow)):
         if f.args[0] == symbol:
             return [f]
+
+        if f.is_Pow:
+            base, expo = f.args
+            if expo.has(symbol):
```

Comment:
The case `(x+1)**y`, fails here because expo does not contain symbol element which is `x`, therefore it does not treats `f` as an exponential function(in terms of `symbol(x)`). Therefore execution continues as usual.

if I try something like:

```
>>> f = (x+1)**y
>>> decompogen(f, y) # decomposing w.r.t `y`.
[(x + 1)**y]  # the result is `f`.
```